tools/PsesPsClient/PsesPsClient.psm1

```diff
@@ -0,0 +1,314 @@
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $HostProfileId = 'TestHost',
+
+        [Parameter()]
+        [ValidateNotNull()]
+        [version]
+        $HostVersion = '1.99',
+
+        [Parameter()]
+        [ValidateNotNull()]
+        [string[]]
+        $AdditionalModules = @('PowerShellEditorServices.VSCode'),
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $BundledModulesPath,
+
+        [Parameter()]
+        [switch]
+        $EnableConsoleRepl
+    )
+
+    $EditorServicesPath = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($EditorServicesPath)
+
+    $instanceId = Get-RandomHexString
+
+    $tempDir = [System.IO.Path]::GetTempPath()
+
+    if (-not $LogPath)
+    {
+        $LogPath = Join-Path $tempDir "pseslogs_$instanceId.log"
+    }
+
+    if (-not $SessionDetailsPath)
+    {
+        $SessionDetailsPath = Join-Path $tempDir "psessession_$instanceId.log"
+    }
+
+    if (-not $BundledModulesPath)
+    {
+        $BundledModulesPath = $script:PsesBundledModulesDir
+    }
+
+    $editorServicesOptions = @{
+        LogPath = $LogPath
+        LogLevel = $LogLevel
+        SessionDetailsPath = $SessionDetailsPath
+        FeatureFlags = $FeatureFlags
+        HostName = $HostName
+        HostProfileId = $HostProfileId
+        HostVersion = $HostVersion
+        AdditionalModules = $AdditionalModules
+        BundledModulesPath = $BundledModulesPath
+        EnableConsoleRepl = $EnableConsoleRepl
+    }
+
+    $startPsesCommand = Unsplat -Prefix "& '$EditorServicesPath'" -SplatParams $editorServicesOptions
+
+    $pwshPath = (Get-Process -Id $PID).Path
+
+    if (-not $PSCmdlet.ShouldProcess("& '$pwshPath' -Command '$startPsesCommand'"))
+    {
+        return
+    }
+
+    $serverProcess = Start-Process -PassThru -FilePath $pwshPath -ArgumentList @(
+        '-NoLogo',
+        '-NoProfile',
+        '-NoExit',
+        '-Command',
+        $startPsesCommand
+    )
+
+    $sessionPath = $editorServicesOptions.SessionDetailsPath
+
+    $i = 0
+    while (-not (Test-Path $sessionPath))
+    {
+        if ($i -ge 10)
+        {
+            throw "No session file found - server failed to start"
+        }
+
+        Start-Sleep 1
+        $null = $i++
+    }
+
+    return [PsesServerInfo]@{
+        PsesProcess = $serverProcess
+        SessionDetails = Get-Content -Raw $editorServicesOptions.SessionDetailsPath | ConvertFrom-Json
+        StartupOptions = $editorServicesOptions
+    }
+}
+
+function Connect-NamedPipe
+{
+    [OutputType([PsesPsClient.LspPipe])]
+    param(
+        [Parameter(Mandatory)]
+        [string]
+        $PipeName
+    )
+
+    $psesIdx = $PipeName.IndexOf('PSES')
+    if ($psesIdx -gt 0)
+    {
+        $PipeName = $PipeName.Substring($psesIdx)
+    }
+
+    $pipe = [PsesPsClient.LspPipe]::Create($PipeName)
+    $pipe.Connect()
+    return $pipe
+}
+
+function Send-LspInitializeRequest
```

Comment:
Would it be worth having a `Send-LspRequest` that would be more generic so that tests could be written on day-1 for any message, rather than needing a subsequent change to this module?

I'm on the fence about this.

Reply:
I actually originally had this, but took it out when I wrote the C# class.

I've put it in now, to go with other PowerShell wrappers I added